### ISSUE TITLE
[Feature Request] Hourly step count?

### ISSUE THREAD

**Issue by andydixon** [jdhartley/terminal-for-fitbit#20]
Love the watch face, and when I get some spare time, i'd love to contribute!

One feature I would find useful would be an hourly step count. I am not sure how easy it'd be to implement, but it would make my hourly step goal easier to understand.

Perhaps [STEP] could look something like:     XXXX steps / (XXX hr)

But granted, space is tight, and I have no talent whatsoever on anything front-endy :D

**Comment by jdhartley** [comment]
Hey @andydixon thanks for the feature request! I love exploring new ideas for Terminal

I am doing a bit of digging into the api for time slices like this. I do some some minutehistory stuff that may be able to help, but I would need to do a bit more digging with a build to see the data I get back. My biggest fear is it being incomplete or inaccurate. Something to test!!

Curious your opinion on the display piece! Would this make sense as a new line of data? Something like `[HRLY] 1234 steps` maybe?

If it's ideal to instead keep it in the same line (I get it, space is tight!), I would probably create a second Step count dataline called "Step count with hourly" and it would display as `[STEP] 12,345 | 1234 hr` or `[STEP] 12,345 (1234 hr)`

**Comment by andydixon** [comment]
Hey,
Personally, I don't mind - Either a separate line or a combination would be pretty cool. Thinking about it though with people being way more active than I am, there could be a potential for wrapping off the edge of the screen if it's all on one line. Maybe having both options will cater for everyone :D

**Comment by pday5** [comment]
I actually just posted this request before realizing someone else already had. I suggested [STEP] total / hourly where hourly is reset to 0 at the top of each hour. So sorry for the redundant request. I will delete the one I just posted.

**Comment by jdhartley** [comment]
Thanks @pday5 — I am digging into this a bit more today. I see I was looking into this _(oh my god 2020 was SIX years ago?!)_ with the `minuteHistory` time slices, and I think I was onto something. Going to play around with some code to get "steps since top of hour excluding current minute".

what this means is you will see live step counts on the left, and then the hourly number will update in chucks at the top of each minute. not Perfect™ but still achieves our goals


For display I am leaning towards:
```
[STEP] 12,345 | 1234 hr
```

I still have a device to test on somewhere in the house, but I am not dailying a fitbit these days. would love help testing on-device once I have some code to share

**Comment by pday5** [comment]
Your display example looks great! I'm happy to help test, if needed. I'm currently using Versa 4.

**Comment by jdhartley** [comment]
I have some code I feel good about, just can't quite test in the simulator. Going to give this a bit more tinkering and testing and then will work on releasing to Fitbit Gallery

Using the sidebeam right now (_mostly_ out of laziness, but also for better spacing). If the separator in the screenshot is ugly to you, let me know and I will look at a different pipe glyph instead!!

```
[STEP] 12,345  ▏ 1234+ hr
```

<img width="1084" height="785" alt="Image" src="https://github.com/user-attachments/assets/a5b2736f-ccc2-456d-8361-9d400247befb" />

**Comment by pday5** [comment]
I like it! I think it looks great!

**Comment by jdhartley** [comment]
I've uploaded [v2.5.0](https://github.com/jdhartley/terminal-for-fitbit/releases/tag/2.5.0) to GAM and it's awaiting review. I will close this issue once published

[v3.1.0](https://github.com/jdhartley/terminal-for-fitbit/releases/tag/3.1.0) is marked as a prerelease and is available for download and sideloading!

if anyone out there is still running a sdk4 device (versa 1/2/lite/ionic) and wants this feature let me know and I will ship backport and ship a legacy build to GAM for you.

If you run into any issues or data glitches please share

**Comment by pday5** [comment]
Once it's approved, is it just a matter of me applying an update through the Fitbit app, or what do I do to get the newest version of your watch face? I'm so pumped you added this feature! It is the only thing I was missing, and I really like the watch face otherwise. Thank you so much for being so responsive and productive!

**Comment by jdhartley** [comment]
@pday5 Yeah! once approved it will just be an update. Hoping Fitbit reviews my app build soon, but if this ends up taking a while and you want to try it out: you can use the fitbit command line tool to connect to your phone and device and install the clockface manually using the fba file from the release
